src/ensembl/src/content/app/species/state/sidebar/speciesSidebarSlice.ts

```diff
@@ -110,7 +116,10 @@ const speciesPageSidebarSlice = createSlice({
       }>
     ) {
       if (!state.species[action.payload.genomeId]) {
-        state.species[action.payload.genomeId] = { payload: null };
+        state.species[action.payload.genomeId] = {
+          payload: null,
+          isSidebarOpen: true
+        };
```

Comment:
There are several strange things going on; I am not sure I am reading them right.

1) `setSidebarPayloadForGenomeId` handles situations when there isn't `state.species[action.payload.genomeId]`, but `toggleSidebar` doesn't. Is this a good thing?
2) The selector's default `isSidebarOpen` value for when there is no active genome id is `false`, but the reducer sets the default value as `true`. Is this inconsistent?
3) More importantly, do we know whether the code as currently written will even work? I see `state.species[action.payload.genomeId]` is first assigned an object value, and then this object is modified. I believe immer shouldn't even allow this to happen — from what I recall, it freezes the modified part of the subtree.

Consider the following alternative:

```ts
import get from 'lodash/get';
import merge from 'lodash/merge';


type StateForGenome = {
  payload: SpeciesSidebarPayload | null;
  isSidebarOpen: boolean;  
};

const initialStateForGenome: StateForGenome = {
  payload: null,
  isSidebarOpen: true
};

const updateStateForGenome = (state: SpeciesPageSidebarState, genomeId: string, fragment: Partial<StateForGenome>) => {
  const stateForGenome = get(state, `species.${genomeId}`, initialStateForGenome);
  const updatedStateForGenome = merge({}, stateForGenome, fragment);
  state.species[genomeId] = stateForGenome; // here we will be relying on the immer magic
};


// then, in the action creator inside the createSlice

    toggleSidebar(
      state,
      action: PayloadAction<{
        genomeId: string;
      }>
    ) {
      const isSidebarOpen = state.species[action.payload.genomeId].isSidebarOpen ?? true;
      updateStateForGenome(state, action.genomeId, { isSidebarOpen });

    }

// etc.
```

Reply:
>setSidebarPayloadForGenomeId handles situations when there isn't state.species[action.payload.genomeId], but toggleSidebar doesn't. Is this a good thing?

I thought the sidebar will not get displayed without the payload but I slightly had a strange feeling about this change. thanks for pointing it out.

>The selector's default isSidebarOpen value for when there is no active genome id is false, but the reducer sets the default value as true. Is this inconsistent? 

Good spot. No, I think the selector should return true by default.

>More importantly, do we know whether the code as currently written will even work? I see state.species[action.payload.genomeId] is first assigned an object value, and then this object is modified. I believe immer shouldn't even allow this to happen — from what I recall, it freezes the modified part of the subtree.

Yes, you are right. I did test it and it was working so I am not sure how. I have now updated it with the suggested alternative. Thanks